SimPEG/EM/Static/DC/IODC.py

```diff
@@ -0,0 +1,444 @@
+        self, a_locations, b_locations, m_locations, n_locations,
+        survey_type=None, dobs=None,
+        data_type="volt", fname=None, dim=2
+    ):
+        """
+        read A, B, M, N electrode location and data (V or appResistivity)
+        """
+        geometric_factor = SimPEG.EM.Static.Utils.StaticUtils.geometric_factor
+
+        self.a_locations = a_locations.copy()
+        self.b_locations = b_locations.copy()
+        self.m_locations = m_locations.copy()
+        self.n_locations = n_locations.copy()
+        self.survey_type = survey_type
+        self.data_type = data_type
+        self.dim = dim
+
+        uniqSrc = Utils.uniqueRows(np.c_[self.a_locations, self.b_locations])
+        uniqElec = SimPEG.Utils.uniqueRows(
+            np.vstack((self.a_locations, self.b_locations, self.m_locations, self.n_locations))
+            )
+        self.electrode_locations = uniqElec[0]
+        nSrc = uniqSrc[0].shape[0]
+        ndata = self.a_locations.shape[0]
+
+        if self.survey_geometry == "SURFACE":
+            # 2D locations
+            srcLists = []
+            sortinds = []
+            for iSrc in range (nSrc):
+                inds = uniqSrc[2] == iSrc
+                sortinds.append(np.arange(ndata)[inds])
+
+                locsM = self.m_locations[inds, :]
+                locsN = self.n_locations[inds, :]
+
+                if (survey_type == 'dipole-dipole') or (survey_type == 'pole-dipole'):
+                    rx = Rx.Dipole_ky(locsM, locsN)
+                elif (survey_type == 'dipole-pole') or (survey_type == 'pole-pole'):
+                    rx = Rx.Pole_ky(locsM)
+
+                if dim == 2:
+                    locA = uniqSrc[0][iSrc, :2]
+                    locB = uniqSrc[0][iSrc, 2:]
+                elif dim == 3:
+                    locA = uniqSrc[0][iSrc, :3]
+                    locB = uniqSrc[0][iSrc, 3:]
+
+                if (survey_type == 'dipole-dipole') or (survey_type == 'dipole-pole'):
+                    src = Src.Dipole([rx], locA, locB)
+                elif (survey_type == 'pole-dipole') or (survey_type == 'pole-pole'):
+                    src = Src.Pole([rx], locA)
+
+                srcLists.append(src)
+
+            self.sortinds = np.hstack(sortinds)
+            survey = Survey_ky(srcLists)
+            self.a_locations = self.a_locations[self.sortinds, :]
+            self.b_locations = self.b_locations[self.sortinds, :]
+            self.m_locations = self.m_locations[self.sortinds, :]
+            self.n_locations = self.n_locations[self.sortinds, :]
+            self.G = geometric_factor(
+                survey, survey_type=self.survey_type,
+                space_type=self.space_type
+                )
+            if dobs is None:
+                self.dobs = 100.*self.G
+            else:
+                self.dobs = dobs[self.sortinds]
+
+            if self.data_type == "volt":
+                self.volt = self.dobs.copy()
```

Comment:
I think volt should be an `@property` (and maybe it is `voltages`, this is an array, right?)
```
@property
voltages(self):
    if self.data_type == "volt":
        return self.dobs
    elif self.data_type == "apparent_resistivity":
        # calculate it

```

Reply:
yep.